src/dynamics/mpas/driver/dyn_mpas_subdriver.F90

```diff
@@ -759,9 +809,290 @@ subroutine dyn_mpas_init_phase3(self, cam_pcnst, pio_file)
+        character(*), parameter :: mpas_scalar_qv_standard_name(*) = [ character(strkind) :: &
+            'water_vapor_mixing_ratio_wrt_dry_air', &
+            'water_vapor_mixing_ratio_wrt_moist_air', &
+            'water_vapor_mixing_ratio_wrt_moist_air_and_condensed_water' &
+        ]
+
+        character(*), parameter :: subname = 'dyn_mpas_subdriver::dyn_mpas_define_scalar'
+        integer :: i, j, ierr
+        integer :: index_qv, index_water_start, index_water_end
+        integer :: time_level
+        type(field3dreal), pointer :: field_3d_real => null()
+        type(mpas_pool_type), pointer :: mpas_pool => null()
+
+        call self % debug_print(subname // ' entered')
+
+        if (self % number_of_constituents == 0) then
+            call self % model_error('Constituents must be allocated before being defined', subname, __LINE__)
+        end if
+
+        ! Input sanitization.
+
+        if (size(constituent_name) /= size(is_water_species)) then
+            call self % model_error('Mismatch between numbers of constituent names and their waterness', subname, __LINE__)
+        end if
+
+        if (size(constituent_name) == 0 .and. self % number_of_constituents == 1) then
+            ! If constituent definitions are empty, `qv` is the only constituent per MPAS requirements.
+            ! See `dyn_mpas_init_phase3` for details.
+            allocate(self % constituent_name(1), stat=ierr)
+
+            if (ierr /= 0) then
+                call self % model_error('Failed to allocate constituent_name', subname, __LINE__)
+            end if
+
+            allocate(self % is_water_species(1), stat=ierr)
+
+            if (ierr /= 0) then
+                call self % model_error('Failed to allocate is_water_species', subname, __LINE__)
+            end if
+
+            self % constituent_name(1) = mpas_scalar_qv_standard_name(1)
+            self % is_water_species(1) = .true.
+        else
+            if (size(constituent_name) /= self % number_of_constituents) then
+                call self % model_error('Mismatch between numbers of constituents and their names', subname, __LINE__)
+            end if
+
+            if (any(len_trim(adjustl(constituent_name)) > len(self % constituent_name))) then
+                call self % model_error('Constituent names are too long', subname, __LINE__)
+            end if
+
+            allocate(self % constituent_name(self % number_of_constituents), stat=ierr)
+
+            if (ierr /= 0) then
+                call self % model_error('Failed to allocate constituent_name', subname, __LINE__)
+            end if
+
+            self % constituent_name(:) = adjustl(constituent_name)
+
+            allocate(self % is_water_species(self % number_of_constituents), stat=ierr)
+
+            if (ierr /= 0) then
+                call self % model_error('Failed to allocate is_water_species', subname, __LINE__)
+            end if
+
+            self % is_water_species(:) = is_water_species
```

Comment:
In my opinion, it makes the code a bit clearer to add `(:)` to `is_water_species`; i.e.,
```fortran
self % is_water_species(:) = is_water_species(:)
```

Reply:
I pushed https://github.com/ESCOMP/CAM-SIMA/pull/267/commits/b2ef93e070812e3b2f9a42ec779865762985e97b to resolve this. Other similar instances in the code also follow this style.